tests/examples/wibox/container/defaults/template.lua

```diff
@@ -0,0 +1,96 @@
+awesome = require( "awesome" )
+client  = require( "client"  )
+tag     = require( "tag"     )
+drawin  = require( "drawin"  )
+screen  = require( "screen"  )
+
+-- Force luacheck to be silent about setting those as unused globals
+assert(awesome and client and tag)
+
+local beautiful = require( "beautiful"     )
+local wibox     = require( "wibox"         )
+local surface   = require( "gears.surface" )
+local shape     = require( "gears.shape"   )
+
+-- If luacov is available, use it. Else, do nothing.
+pcall(function()
+    require("luacov.runner")(luacovpath)
+end)
+
+-- Let the test request a size and file format
+local before, after = loadfile(file_path)()
+
+local container = wibox.widget {
+    {
+        {
+            {
+                before,
+                shape_border_color = beautiful.border_color,
+                shape_border_width = beautiful.border_width,
+                shape              = shape.rounded_rect,
+                widget             = wibox.container.background,
+            },
+            strategy = 'exact',
+            width    = 70,
+            height   = 40,
+            widget   = wibox.container.constraint
+        },
+        {
+            {
+                {
+                    text   = " ",
+                    widget = wibox.widget.textbox,
+                },
+                bg                 = beautiful.bg_normal,
+                shape_border_color = beautiful.border_color,
+                shape_border_width = beautiful.border_width,
+                widget             = wibox.container.background,
+                shape              = shape.transform(shape.arrow)
+                                        : rotate_at(15,15,math.pi/2)
+                                        : translate(0,-8)
+                                        : scale(0.9, 0.9),
+            },
+            strategy = 'exact',
+            width    = 30,
+            height   = 30,
+            widget   = wibox.container.constraint
+        },
+        {
+            {
+                after,
+                shape_border_color = beautiful.border_color,
+                shape_border_width = beautiful.border_width,
+                shape              = shape.rounded_rect,
+                widget             = wibox.container.background,
+            },
+            strategy = 'exact',
+            width    = 70,
+            height   = 40,
+            widget   = wibox.container.constraint
+        },
+        layout = wibox.layout.align.horizontal
+    },
+    margins = 10,
+    widget  = wibox.container.margin,
+}
+
+-- Emulate the event loop for 10 iterations
+for _ = 1, 10 do
+    awesome:emit_signal("refresh")
+end
+
+-- Get the example fallback size (the tests can return a size if the want)
+local f_w, f_h = container:fit({dpi=96}, 9999, 9999)
+
+-- There is an overhead that cause testbox "...", add 10 until someone
+-- figures out the real equation
+f_w, f_h = f_w+10, f_h+10
```

Comment:
Uhm, all textboxes are inside of `constraint` containers, so their `fit` result is ignored. I guess a single space is just wider than how much space you give it here? My suggested fix would be to use the empty string instead of a single space. Even better would be not to use a textbox at all since you really only want a widget doing nothing. I thought the declarative layout allows to construct widgets on the fly? All callbacks should be optional by now, so could that be used instead of a textbox here?

btw: spelling: testbox -> textbox